### ISSUE TITLE
azurerm_mssql_managed_instance "data" element does not reflect all attributes set in "resource" elment

### ISSUE THREAD

**Issue by Haratsu** [hashicorp/terraform-provider-azurerm#23268]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.7

### AzureRM Provider Version

3.70.00

### Affected Resource(s)/Data Source(s)

azurerm_mssql_managed_instance

### Terraform Configuration Files

```hcl
data "azurerm_mssql_managed_instance" "existing" {
  name                           = local.instanceName
  resource_group_name            = var.resourceGroupName
}

resource "null_resource" "instance_check" {
  triggers = {
    missing               = data.azurerm_mssql_managed_instance.existing == null
    desired_config_changed = tolower(data.azurerm_mssql_managed_instance.existing.maintenance_configuration_name) != tolower(local.maintenanceConfigurationName)
  }
}
```

### Debug Output/Panic Output

```shell
This object has no argument, nested block, or exported attribute named "maintenance_configuration_name".
```

### Expected Behaviour

data source and resource do have the same attributes

### Actual Behaviour

As you can see in the docu (link below) of resource `azurerm_mssql_managed_instance` there is an the attribute `maintenance_configuration_name` to be configurable
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_managed_instance#maintenance_configuration_name
Unfortunately neither documentation nor actual code for data source of `azurerm_mssql_managed_instance` provide this attribute:
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/mssql_managed_instance

The issue came to my attention as when using the resource documentation on the attribute. It shall use the pattern `"SQL_{Location}_MI_{Size}"`. 
When using the attribute `"location"` of the resource group that contains the resource as an input for the location placeholder in the format the value of location will be all lower case. Which works during initial creation of the resource.
Once the resource is created Azure changes the value to some "random" case. E.g.
- resource group location `"switzerlandnorth"`
- azurerm_mssql_managed_instance: = `"SQL_${azurerm_resource_group.example.location}_MI_2"`
- tfplan output of second run:
```
 ~ resource "azurerm_mssql_managed_instance" "main" {
        id                             = "/subscriptions/27a79318-4a38-4987-bca1-991e9adf4d6d/resourceGroups/rg-az007-ch01-shar-data-tst/providers/Microsoft.Sql/managedInstances/sql-managed-db-1"
      ~ maintenance_configuration_name = "SQL_SwitzerlandNorth_MI_1" -> "SQL_switzerlandnorth_MI_1"
```

I did not want to bother anyone with "extending" the change detection for this attribute to ignore cases. Also did not want to create a map of real location names to "random"-case location names that no-one can maintain. so I introduced some logic using a data source for `azurerm_mssql_managed_instance`and a null resource in combination with a dynamically added lifecycle block that ignores changes if they only contain differences in the case. but that failed as the attribute in question is not available

So basically this issue is about adding all attributes to data source that are available in resource as well. Feel free to create a second ticket about handling Azure's random-case names if needed. In that case I will provide additional code snippets for reproduction.

**Comment by rcskosir** [comment]
@Haratsu Thank you for taking the time to open this feature request!
